packages/compiler/src/lib/decorators.ts

```diff
@@ -343,6 +344,83 @@ export function isErrorModel(program: Program, target: Type): boolean {
   return false;
 }
 
+// -- @mediaTypeHint decorator --------------
+
+const [_getMediaTypeHint, setMediaTypeHint] = useStateMap<MediaTypeHintable, string>(
+  createStateSymbol("mediaTypeHint"),
+);
+
+/**
+ * A type that can have a default MIME type.
+ */
+type MediaTypeHintable = Parameters<MediaTypeHintDecorator>[1];
+
+export const $mediaTypeHint: MediaTypeHintDecorator = (
+  context: DecoratorContext,
+  target: MediaTypeHintable,
+  mediaType: string,
+) => {
+  validateDecoratorUniqueOnNode(context, target, $mediaTypeHint);
+
+  setMediaTypeHint(context.program, target, mediaType);
```

Comment:
can we validate the media type is valid, I think we already have all the logic from encodedName decorator